File: Python_Codes/8. IO Streams/CharacterStreamsExample.py
def write_and_read_file(file_path, content):
    # Write content to the file
    try:
        with open(file_path, 'w', encoding='utf-8') as writer:
            writer.write(content)
            print("Content written to file!")
    except IOError as e:
        print("Error writing to file:", e)

    # Read content from the file
    try:
        with open(file_path, 'r', encoding='utf-8') as reader:
            print("Reading content from file:")
            while True:
                char_data = reader.read(1)  # Read one character at a time
                if not char_data:
                    break
                print(char_data, end='')
            print()
    except IOError as e:
        print("Error reading from file:", e)


if __name__ == "__main__":
    write_and_read_file("C:/WorkSpace/Python_Codes/8. IO Streams/example.txt", "Hello, world!")

File: Projects/JavaProject/src/com/IO_Streams/CharacterStreamsExample.java
package com.IO_Streams;

import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

public class CharacterStreamsExample {
    public static void writeAndReadFile(String filePath, String content) {
        try (FileWriter writer = new FileWriter(filePath)) {
            writer.write(content);
            System.out.println("Content written to file!");
        } catch (IOException e) {
            System.out.println("Error writing to file: " + e.getMessage());
        }

        try (FileReader reader = new FileReader(filePath)) {
            int charData;
            System.out.println("Reading content from file:");
            while ((charData = reader.read()) != -1) {
                System.out.print((char) charData);
            }
            System.out.println();
        } catch (IOException e) {
            System.out.println("Error reading from file: " + e.getMessage());
        }
    }

    public static void main(String[] args) {
        writeAndReadFile("C:\\Java_Codes\\8. IO Streams\\example.txt", "Hello, world!");
    }
}